tests/UnitTests/Benchmark/BenchmarkTest.h

```diff
@@ -0,0 +1,153 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#pragma once
+
+#include <TestFramework.h>
+#include <type_traits>
+#include <chrono>
+#include <numeric>
+#include <utility>
+#include <memory>
+
+namespace testing {
+class BenchmarkTestBase {
+public:
+    inline void Run();
+
+    // Number of times to run each test, default to 1
+    size_t GetRunCount() const {
+        return 1;
+    }
+
+    // Pre-run and Post-run equivalent to SetUp and TearDown per test case
+    // Setup and Tear down that is performed once per all tests should be done in the constructor/destructor
+    void PreRun() {
+        // Default do nothing
+    }
+
+    void PostRun() {
+        // Default do nothing
+    }
+};
+
+namespace internal {
+
+using Seconds = double;
+
+// Helper method shared between BenchmarkTestRunner classes
+// This is called to actually run and benchmark the tests
+// Can't be shared internally because parameterized tests need a different inheritance than non
+template <typename T>
+static void __RunTest(T& test, std::vector<Seconds>& results) {
+    size_t runCount = test.GetRunCount();
+    for (size_t i = 0; i < runCount; ++i) {
+        test.PreRun();
+        auto start = std::chrono::high_resolution_clock::now();
+        test.Run();
+        auto end = std::chrono::high_resolution_clock::now();
+        test.PostRun();
+        std::chrono::duration<Seconds> duration = end - start;
```

Comment:
What was the verdict on this?

---
In reply to: [98574225](https://github.com/Microsoft/WinObjC/pull/1876#discussion_r98574225) [](ancestors = 98574225)